### ISSUE TITLE
Cannot set options in aggregate pre hook

### ISSUE THREAD

**Issue by clhuang** [Automattic/mongoose#7657]
**Do you want to request a *feature* or report a *bug*?**
Unclear what expected behavior is

**What is the current behavior?**
Aggregate options are copied before running the pre hook (https://github.com/Automattic/mongoose/blob/5.4.20/lib/aggregate.js#L920), so that any modifications to the options in the pre hook are dropped

In model file:
```js
schema.pre('aggregate', function() {
  this.option({ maxTimeMS: 30000 });
}
```
elsewhere:
```js
await Model.aggregate([{ $match: { unindexedField: 'this will take a while' } }]);
```

The call to `Model.aggregate` does not time out, and calling `db.currentOps` from MongoDB shell would show the aforementioned aggregate query containing option `maxTimeMS: 30000`

**What is the expected behavior?**
Expect that the aggregate options as reported by MongoDB would contain `maxTimeMS: 30000`, and query would time out

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Node 8.15.1, Mongoose 5.4.20, MongoDB 3.4

**Comment by stevenhao** [comment]
i am also experiencing this

**Comment by C-Duv** [comment]
@stevenhao #7606 is fixed in v5.4.22 (I just checked).
